DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/svd/src/memory_transfers.hpp

```diff
@@ -0,0 +1,172 @@
+#ifndef __MEMORY_TRANSFERS_HPP__
+#define __MEMORY_TRANSFERS_HPP__
+
+#include "constexpr_math.hpp"
+#include "tuple.hpp"
+#include "unrolled_loop.hpp"
+
+/*
+  Read matrix_count matrices of type TT from DDR by bursts of num_elem_per_bank
+  elements, and write the matrices to the "MatrixPipe" pipe num_elem_per_bank by
+  num_elem_per_bank elements.
+  Repeat this operations "repetitions" times.
+*/
+template <typename TT,            // Datatype of the elements of the matrix
+          int rows,               // Number of rows of the matrix
+          int columns,            // Number of columns of the matrix
+          int num_elem_per_bank,  // Number of TT elements per DDR burst access
+          typename MatrixPipe     // Output matrix pipe
+          >
+void MatrixReadFromDDRToPipeByBlocks(
+    TT* matrix_ptr,    // Input matrix pointer
+    int matrix_count,  // Number of matrix to read from DDR
+    int repetitions    // Number of time to write the same matrix to the pipe
```

Comment:
```suggestion
    int repetitions    // Number of time to write the same matrix to the pipe
```
in 'real life' I can't see anyone ever using a value other than `1`, as they wouldn't generally want to process the same matrix multiple times.

Reply:
This is useful for performance evaluation: we want to run a very large number of matrices through the kernel to get an accurate TP measurement. But are also limited by the DDR on the FPGA.
Using this `repetition` we can generate a handful for matrices (limiting the memory footprint) and repeat the compute many times, giving us an accurate TP value.